src/Interpreters/Context.h

```diff
@@ -25,6 +25,7 @@
 #include <Backups/BackupsInMemoryHolder.h>
 
 #include <Poco/AutoPtr.h>
+#include <Poco/BinaryWriter.h>
```

Comment:
There's no need to include this header if all we want to do is to forward-declare `BinaryWriter`

Reply:
>adding any character to the source file would invalidate the cache for this TU

Not really, `chcache` calls preprocessor with `-fminimize-whitespace`, so the change needs to be meaningful.